sdk/python/feast/infra/online_stores/astra.py

```diff
@@ -0,0 +1,297 @@
+            table_name = _table_id(project, table)
+
+            # Create table if not exist
+            cql_create_table = _create_cql_table(key_space, table_name,
+                                                 primary_key=["entity_key", "feature_name"],
+                                                 columns=[
+                                                     "entity_key",
+                                                     "feature_name",
+                                                     "value",
+                                                     "event_ts",
+                                                     "created_ts"
+                                                 ],
+                                                 column_types=[
+                                                     "BLOB",
+                                                     "TEXT",
+                                                     "BLOB",
+                                                     "timestamp",
+                                                     "timestamp"
+                                                 ]
+                                                 )
+            self._session = self._initialize_astra_session(config.online_store)
+            self._session.execute(cql_create_table)
+
+            # Now create Index
+            cql_index = _create_index_cql(key_space, table_name+"_ek", table_name, "entity_key")
+            self._session.execute(cql_index)
+
+        for table in tables_to_delete:
+            table_name = _table_id(project, table)
+            delete_cql = _create_delete_table_cql(key_space, table_name)
+            self._session.execute(delete_cql)
+
+    def teardown(
+        self,
+        config: RepoConfig,
+        tables: Sequence[Union[FeatureTable, FeatureView]],
+        entities: Sequence[Entity],
+    ):
+        online_config = config.online_store
+        assert isinstance(online_config, AstraConfig)
+        key_space = online_config.keyspace
+        for table in tables:
+            table_name = _table_id(config.project, table)
+            cql_delete = _create_delete_table_cql(key_space, table_name)
+            self._session = self._initialize_astra_session(config.online_store)
+            self._session.execute(cql_delete)
+
+
+def _create_cql_table(key_space: str,
+                      table_name: str,
+                      primary_key: List,
+                      columns: List[str],
+                      column_types: List[str]
+                      ) -> str:
+    """
+    in this function we will create a CQL to create a table
+    """
+    assert len(columns) > 0, "Columns can not be empty in a Table. "
+
+    assert len(columns) == len(column_types), "Length of columns and type of columns length missmatch."
+
+    # If we have same amount of columns and their types
+    cql_create_table = "CREATE TABLE  IF NOT EXISTS " + key_space + "." + table_name

```

Comment:
looks like there's an extra space between TABLE and IF?

Reply:
yes, got it thanks